core/vm/contracts_stateful.go

```diff
@@ -171,6 +173,67 @@ func (c *nativeAssetCall) Run(evm *EVM, caller ContractRef, addr common.Address,
 	return ret, remainingGas, err
 }
 
+// exportAsset atomically transfers a native asset to a recipient address as well as calling that
+// address
+type exportAsset struct {
+	gasCost uint64
+}
+
+func PackExportAssetInput(address common.Address, assetID common.Hash, assetAmount *big.Int, callData []byte) []byte {
+	input := make([]byte, 84+len(callData))
+	copy(input[0:20], address.Bytes())
+	copy(input[20:52], assetID.Bytes())
+	assetAmount.FillBytes(input[52:84])
+	copy(input[84:], callData)
+	return input
+}
+
+func UnpackExportAssetInput(input []byte) (string, string, *big.Int, error) {
+	if len(input) < 109 {
+		return "", "", nil, fmt.Errorf("export asset call input had unexpcted length %d", len(input))
+	}
+
+	recipient := string(input[:45]) // string encoded X-* address.
+
+	// TODO: confirm an encoded assetID is 32 bytes in length
+	assetID := string(common.TrimLeftZeroes(input[45:77])) // string encoded assetID. could be 0 left padded literal "AVAX".
```

Comment:
Unpack the string encoded assetID. Questions:

1. Is this the correct way to do it?
2. What is the length of a string encoded asset ID? And is it always that length?